infrastructure/zk/src/database.ts

```diff
@@ -1,101 +1,233 @@
 import { Command } from 'commander';
 import * as utils from './utils';
 
-export async function reset() {
+export async function reset(opts: any) {
     await utils.confirmAction();
-    await wait();
-    await drop();
-    await setup();
+    await wait(opts);
+    await drop(opts);
+    await setup(opts);
 }
 
-export async function resetTest() {
-    process.env.DATABASE_URL = process.env.TEST_DATABASE_URL;
-    await utils.confirmAction();
+enum DalPath {
+    CoreDal = 'core/lib/dal',
+    ProverDal = 'prover/prover_dal'
+}
+
+function getDals(opts: any): Map<DalPath, string> {
+    let dals = new Map<DalPath, string>();
+    if (!opts.prover && !opts.server) {
+        dals.set(DalPath.CoreDal, process.env.DATABASE_URL!);
+        dals.set(DalPath.ProverDal, process.env.DATABASE_PROVER_URL!);
+    }
+    if (opts.prover) {
+        dals.set(DalPath.ProverDal, process.env.DATABASE_PROVER_URL!);
+    }
+    if (opts.server) {
+        dals.set(DalPath.CoreDal, process.env.DATABASE_URL!);
+    }
+    return dals;
+}
+
+function getTestDals(opts: any): Map<DalPath, string> {
+    let dals = new Map<DalPath, string>();
+    if (!opts.prover && !opts.server) {
+        dals.set(DalPath.CoreDal, process.env.TEST_DATABASE_URL!);
+        dals.set(DalPath.ProverDal, process.env.TEST_DATABASE_PROVER_URL!);
+    }
+    if (opts.prover) {
+        dals.set(DalPath.ProverDal, process.env.TEST_DATABASE_PROVER_URL!);
+    }
+    if (opts.server) {
+        dals.set(DalPath.CoreDal, process.env.TEST_DATABASE_URL!);
+    }
+    return dals;
+}
```

Comment:
Generally speaking, I feel like this whole file became a bit boilerplate-ish (if that's even a word).
We deal with a set of `{server, prover} x {test, non_test}` here for which we maintain some currently implicit state (exposed by these methods and by `xxx` and `xxxTest` function variants) and I feel like we can express it better by having something like (pseudocode):

```typescript
export enum DBType { Server, Prover };

export class DBManager {
    public constructor(type: DBType, test: boolean) {  /* Load required envs, setup paths, etc */ }

    public async drop() { .. }
    public async migrate() { .. }
    // .. whatever
}
```

Reply:
You can always wrap stuff into a closure for commander's sake, no?
E.g. `.action((opts) => (new DBManager(opts, false)).drop())` or something like that.